extension/source/Controllers/Network/createJsonRpcClient.ts

```diff
@@ -46,6 +49,85 @@ export function createProviderConfigMiddleware(
   };
 }
 
+function mockGetTransactionByHashMiddleware(): JRPCMiddleware<
+  unknown,
+  unknown
+> {
+  return (
+    req: JRPCRequest<unknown>,
+    res: JRPCResponse<unknown>,
+    next: JRPCEngineNextCallback,
+    end: JRPCEngineEndCallback,
+  ) => {
+    if (req.method === 'eth_getTransactionByHash') {
+      const hash: string = (req.params as any)[0];
+
+      if (hash in knownTransactions) {
+        const knownTx = knownTransactions[hash];
+
+        // Here we're just relaying the information that we already know
+        // internally about the transaction. ethers needs this response to
+        // function properly.
+        res.result = {
+          hash,
+          from: knownTx.from,
+          nonce: knownTx.nonce,
+          value: knownTx.value,
+          gasLimit: '0x0',
+          data: knownTx.data,
+        };
+
+        return end();
+      }
+    }
+
+    return next();
+  };
+}
+
+function createAggregatorMiddleware(): JRPCMiddleware<unknown, unknown> {
+  return async (
+    req: JRPCRequest<unknown>,
+    res: JRPCResponse<unknown>,
+    next: JRPCEngineNextCallback,
+    end: JRPCEngineEndCallback,
+  ) => {
+    if (req.method === 'eth_getTransactionReceipt') {
+      const hash: string = (req.params as any)[0];
+
+      if (hash in knownTransactions) {
+        const knownTx = knownTransactions[hash];
+
+        const aggregator = new Aggregator(AGGREGATOR_URL);
+        const bundleReceipt = await aggregator.lookupReceipt(hash);
+
+        if (bundleReceipt === undefined) {
+          res.result = null;
+          return end();
+        }
+
+        res.result = {
+          transactionHash: hash,
+          transactionIndex: bundleReceipt.transactionIndex,
+          blockHash: bundleReceipt.blockHash,
+          blockNumber: bundleReceipt.blockNumber,
+          from: knownTx.from,
+          to: knownTx.to,
+          logs: [],
+          cumulativeGasUsed: '0x0',
+          gasUsed: '0x0',
+          status: '0x1',
+          effectiveGasPrice: '0x0',
```

Comment:
Can we included gas stats? Likely split to another issue.

Reply:
Need to think a bit more about this. don't think it's currently worth pursuing in the near/medium future.